From: Emanuele Rocca <ema@wikimedia.org>
Date: Sat, 16 Jan 2016 15:39:45 +0100
Subject: varnishd-persistent-mmap-exit

Detect and exit when persistent storage can't mmap a file at the
required address.
---
 bin/varnishd/storage/mgt_storage_persistent.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/bin/varnishd/storage/mgt_storage_persistent.c b/bin/varnishd/storage/mgt_storage_persistent.c
index 9ea6047..b311ba3 100644
--- a/bin/varnishd/storage/mgt_storage_persistent.c
+++ b/bin/varnishd/storage/mgt_storage_persistent.c
@@ -218,6 +218,11 @@ smp_mgt_init(struct stevedore *parent, int ac, char * const *av)
 		ARGV_ERR("(-spersistent) failed to mmap (%s)\n",
 		    strerror(errno));
 
+	if (target != NULL && sc->base != target) {
+		fprintf(stderr, "Could not mmap SILO (%s) at target %p, was mapped at %p instead\n", sc->filename, target, sc->base);
+		exit(75); /* EX_TEMPFAIL */
+	}
+
 	smp_def_sign(sc, &sc->idn, 0, "SILO");
 	sc->ident = SIGN_DATA(&sc->idn);
 
